airflow/traces/otel_tracer.py

```diff
@@ -0,0 +1,333 @@
+        """
+        return tracer
+
+    def get_current_span(self):
+        return trace.get_current_span()
+
+    def use_span(self, span: Span):
+        return trace.use_span(span=span)
+
+    def start_span(
+        self,
+        span_name: str,
+        component: str | None = None,
+        parent_sc: SpanContext | None = None,
+        span_id=None,
+        links=None,
+        start_time=None,
+    ):
+        """Start a span. if service_name is not given, otel_service is used."""
+        if component is None:
+            component = self.otel_service
+
+        trace_id = self.get_current_span().get_span_context().trace_id
+        if span_id is not None:
+            tracer = self.get_tracer_with_id(component=component, trace_id=trace_id, span_id=span_id)
+        else:
+            tracer = self.get_tracer(component)
+
+        kvs = {}
+        if self.tags is not None:
+            kvs = parse_tracestate(self.tags)
+
+        if links is not None:
+            _links = gen_links_from_kv_list(links)
+        else:
+            _links = []
+
+        if start_time is not None:
+            start_time = int(start_time.timestamp() * 1000000000)
+
+        if parent_sc is not None:
+            ctx = trace.set_span_in_context(NonRecordingSpan(parent_sc))
+            span = tracer.start_as_current_span(
+                span_name, context=ctx, attributes=kvs, links=_links, start_time=start_time
+            )
+        else:
+            span = tracer.start_as_current_span(
+                span_name, attributes=kvs, links=_links, start_time=start_time
+            )
+        return span
+
+    def start_span_from_dagrun(
+        self, dagrun, span_name: str | None = None, component: str = "dagrun", links=None
+    ):
+        """Produce a span from dag run."""
+        # check if dagrun has configs
+        conf = dagrun.conf
+        trace_id = int(gen_trace_id(dag_run=dagrun), 16)
+        span_id = int(gen_dag_span_id(dag_run=dagrun), 16)
+
+        if conf is not None:
+            traceparent = conf.get(TRACEPARENT)
+            tracestate = conf.get(TRACESTATE)
+
+        tracer = self.get_tracer_with_id(component=component, span_id=span_id, trace_id=trace_id)
+
+        kvstr = None
+        # merge attributes from tags and tracestate
+        if self.tags is not None:
+            kvstr = self.tags
+        if tracestate is not None:
+            if kvstr is None:
+                kvstr = tracestate
+            else:
+                kvstr = kvstr + "," + tracestate
+        kvs = parse_tracestate(kvstr)
+
+        if span_name is None:
+            span_name = dagrun.dag_id
+
+        if links is not None:
+            _links = gen_links_from_kv_list(links)
+        else:
+            _links = []
+
+        a_link = Link(
+            context=trace.get_current_span().get_span_context(),
+            attributes={"meta.annotation_type": "link", "from": "parenttrace"},
+        )
+        _links.append(a_link)
```

Comment:
I'd just combine this, I don't see a reason to expand this

```suggestion
        _links.append(
            Link(
                context=trace.get_current_span().get_span_context(),
                attributes={"meta.annotation_type": "link", "from": "parenttrace"},
             )
         )
```